Sources/GRPCReflectionService/Server/ReflectionService.swift

```diff
@@ -425,3 +437,43 @@ where Success == Grpc_Reflection_V1_ServerReflectionResponse.OneOf_MessageRespon
     }
   }
 }
+
+@available(macOS 10.15, iOS 13, tvOS 13, watchOS 6, *)
+extension ReflectionService {
+  static func readSerializedFileDescriptorProto(
+    atPath path: String
+  ) throws -> Google_Protobuf_FileDescriptorProto {
+    let fileURL: URL
+    #if os(Linux)
+    fileURL = URL(fileURLWithPath: path)
+    #else
+    if #available(macOS 13.0, iOS 16.0, tvOS 16.0, watchOS 9.0, *) {
+      fileURL = URL(filePath: path, directoryHint: .notDirectory)
+    } else {
+      fileURL = URL(fileURLWithPath: path)
+    }
+    #endif
+    let binaryData = try Data(contentsOf: fileURL)
+    guard let serializedData = Data(base64Encoded: binaryData) else {
+      throw GRPCStatus(
+        code: .invalidArgument,
+        message:
+          """
+          The \(fileURL.lastPathComponent) file contents could not be transformed \
+          into serialized data representing a file descriptor proto.
+          """
+      )
+    }
+    return try Google_Protobuf_FileDescriptorProto(serializedData: serializedData)
+  }
+
+  static func readSerializedFileDescriptorProtos(
+    atPaths paths: [String]
+  ) throws -> [Google_Protobuf_FileDescriptorProto] {
+    var fileDescriptorProtos = [Google_Protobuf_FileDescriptorProto]()
```

Comment:
We can reserve the capacity of this array as we know how bit it'll be (this lets us avoid reallocating the array on append which could happen multiple times).
```suggestion
    var fileDescriptorProtos = [Google_Protobuf_FileDescriptorProto]()
    fileDescriptorProtos.reserveCapacity(paths.count)

```